server/src/main/java/org/elasticsearch/indices/SystemIndexDescriptor.java

```diff
@@ -432,6 +454,14 @@ public SystemIndexDescriptor getDescriptorCompatibleWith(Version version) {
         return null;
     }
 
+    /**
+     * @return The names of thread pools that should be used for operations on this
+     *    system index.
+     */
+    public SystemIndices.DescriptorThreadPoolNames getThreadPoolNames() {
```

Comment:
I think the `DescriptorThreadPoolNames` class should be defined in the descriptor and not the `SystemIndices` class or just make it its own class.

Reply:
I'll make it its own class. It doesn't seem to fit very naturally as an inner class.